array.go

```diff
@@ -2371,6 +2371,22 @@ func (a *Array) Type() TypeInfo {
 	return nil
 }
 
+func (a *Array) SetType(typeInfo TypeInfo) error {
+	extraData := a.root.ExtraData()
+	extraData.TypeInfo = typeInfo
+
+	a.root.SetExtraData(extraData)
+
+	// Store modified root slab in storage since typeInfo is part of extraData stored in root slab.
+	err := a.Storage.Store(a.root.Header().id, a.root)
```

Comment:
Can this be just:
```suggestion
	err := a.Storage.Store(a.root.ID(), a.root)
```

If so, also the log statement below should be adjusted